synapse/config/metrics.py

```diff
@@ -23,12 +32,30 @@ def read_config(self, config):
         self.metrics_port = config.get("metrics_port")
         self.metrics_bind_host = config.get("metrics_bind_host", "127.0.0.1")
 
+        self.sentry_enabled = "sentry" in config
+        if self.sentry_enabled:
```

Comment:
wouldn't it make more sense to enable sentry if the dsn is set, rather than if the section is present?

Reply:
I'm not a fan of the magic "its enabled if $field is set", as its quite unintuitive (at least to me), especially if you misspell "dsn" or whatever. We could easily enough add a "enabled" field?

Reply:
well, I don't mind tbh. but if you want to stick with doing it this way can you sanity-check that dsn is set rather than throwing a KeyError?